site/zenodo_rdm/config.py

```diff
@@ -335,20 +335,18 @@
 REDIRECTOR_RULES.update(EXPORT_REDIRECTS)
 
 
-def lock_edit_record_published_files(service, identity, record=None):
+def lock_edit_record_published_files(service, identity, record=None, draft=None):
     """Custom conditions for file bucket lock."""
+    # Check if draft already has pids or not, otherwise pass the record
+    use_draft = draft and draft.pids is not None
```

Comment:
Can't we always pass the record? Since this permission is mainly checked when we want to edit an already published record